src/Routes.js

```diff
@@ -90,48 +90,61 @@ const PatchRoutes = () => {
         }
     }, [hasSystems]);
 
-    return (
+    return !hasSystems ? (
         <Suspense
             fallback={
                 <Bullseye>
                     <Spinner />
                 </Bullseye>
             }
         >
-            <Routes>
-                <Route path='*' element={
-                    <AsyncComponent
-                        appId="content_management_zero_state"
-                        appName="dashboard"
-                        module="./AppZeroState"
-                        scope="dashboard"
-                        ErrorComponent={<div>Error state</div>}
-                        app="Content_management"
-                        customFetchResults={ hasSystems }
-                    >
-                        <Routes>
-                            <Route element={<PermissionRoute requiredPermissions={generalPermissions} />}>
-                                <Route path='/advisories' element={<Advisories />} />
-                                <Route path='/advisories/:advisoryId/:inventoryId'
-                                    element={<NavigateToSystem />} />
-                                <Route path='/advisories/:advisoryId' element={<AdvisoryPage />} />
-                                <Route path='/systems' element={<Systems />} />
-                                <Route path='/systems/:inventoryId' element={<InventoryDetail />} />
-                                <Route path='/packages' element={<PackagesPage />} />
-                                <Route path='/packages/:packageName' element={<PackageDetail />} />
-                                <Route path='/packages/:packageName/:inventoryId'
-                                    element={<NavigateToSystem />} />
-                                <Route path='/templates' element={<Templates />} />
-                                <Route path='/templates/:templateName' element={<TemplateDetail />} />
-                                <Route path='*' element={<Navigate to="advisories" />} />
-                            </Route>
-                        </Routes>
-
-                    </AsyncComponent>
-                } />
-
-            </Routes>
+            <AsyncComponent
+                appId="content_management_zero_state"
+                appName="dashboard"
+                module="./AppZeroState"
+                scope="dashboard"
+                app="Content_management"
+                customFetchResults={hasSystems}
+            />
         </Suspense>
+    ) : (
+        <Routes>
+            <Route
+                element={
+                    <PermissionRoute requiredPermissions={generalPermissions} />
```

Comment:
I'm not to sure about doing the permission handling this way. I'd rather have a plain component that renders or not renders the children passed, rather than utilising react-router and it's "Outlet".

Reply:
Yeah, completely agree. Refactored in the last commit.